engine/src/main/java/org/pentaho/di/repository/filerep/KettleFileRepository.java

```diff
@@ -0,0 +1,1571 @@
+            }
+          }
+        }
+      }
+
+      return list;
+    } catch ( Exception e ) {
+      throw new KettleException( "Unable to get list of jobs in folder with id : " + id_directory, e );
+    }
+  }
+
+  public int getNrSubDirectories( ObjectId id_directory ) throws KettleException {
+
+    return 0;
+  }
+
+  @Override
+  public SlaveServer loadSlaveServer( ObjectId id_slave_server, String versionName ) throws KettleException {
+    try {
+      return new SlaveServer( loadNodeFromXML( id_slave_server, SlaveServer.XML_TAG ) );
+    } catch ( Exception e ) {
+      throw new KettleException( "Unable to load slave server from the file repository", e );
+    }
+  }
+
+  @Override
+  public TransMeta loadTransformation( String transname, RepositoryDirectoryInterface repdir,
+    ProgressMonitorListener monitor, boolean setInternalVariables, String versionName ) throws KettleException {
+
+    // This is a standard load of a transformation serialized in XML...
+    //
+    String filename = calcDirectoryName( repdir ) + transname + ".ktr";
+    TransMeta transMeta = new TransMeta( filename, this, setInternalVariables );
+    transMeta.setRepository( this );
+    transMeta.setMetaStore( getMetaStore() );
+    transMeta.setFilename( null );
+    transMeta.setName( transname );
+    transMeta.setObjectId( new StringObjectId( calcObjectId( repdir, transname, EXT_TRANSFORMATION ) ) );
+
+    readDatabases( transMeta, true );
+    transMeta.clearChanged();
+
+    return transMeta;
+  }
+
+  /**
+   * Read all the databases from the repository, insert into the has databases object, overwriting optionally
+   *
+   * @param TransMeta
+   *          The transformation to load into.
+   * @param overWriteShared
+   *          if an object with the same name exists, overwrite
+   * @throws KettleException
+   */
+  public void readDatabases( HasDatabasesInterface transMeta, boolean overWriteShared ) throws KettleException {
+    try {
+      ObjectId[] dbids = getDatabaseIDs( false );
+      for ( int i = 0; i < dbids.length; i++ ) {
+        DatabaseMeta databaseMeta = loadDatabaseMeta( dbids[i], null ); // reads last version
+        if ( transMeta instanceof VariableSpace ) {
+          databaseMeta.shareVariablesWith( (VariableSpace) transMeta );
+        }
+
+        DatabaseMeta check = transMeta.findDatabase( databaseMeta.getName() ); // Check if there already is one in the
+                                                                               // transformation
+        if ( check == null || overWriteShared ) { // We only add, never overwrite database connections.
+          if ( databaseMeta.getName() != null ) {
+            transMeta.addOrReplaceDatabase( databaseMeta );
+            if ( !overWriteShared ) {
+              databaseMeta.setChanged( false );
+            }
+          }
+        }
+      }
+    } catch ( KettleException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)